net/decentralized_dns/constants.h

```diff
@@ -6,21 +6,72 @@
-    ".crypto",     ".x",          ".nft",     ".dao",         ".wallet",
-    ".blockchain", ".bitcoin",    ".zil",     ".altimist",    ".anime",
-    ".klever",     ".manga",      ".polygon", ".unstoppable", ".pudgy",
-    ".tball",      ".stepn",      ".secret",  ".raiin",       ".pog",
-    ".clay",       ".metropolis", ".witg",    ".ubu",         ".kryptic",
-    ".farms",      ".dfz",        ".kresus",  ".binanceus",   ".austin",
-    ".bitget",     ".wrkx"};
+// A struct to be used to guide the lookup of domain suffixes in the fixed flat
+// set below. Lookups using this wrapper will match based on the suffix after
+// the last '.' in the domain. Lookups for suffix can be done like:
+//
+// kUnstoppableDomains.contains(DomainSuffixLookup("https://foo.crypto"))
+//
+struct DomainSuffixLookup {
+  constexpr explicit DomainSuffixLookup(std::string_view domain)
+      : domain(ExtractSuffix(domain)) {}
+
+  // Used to initialise the view from the last '.' in the domain, if any exist.
+  // Otherwise, passes the whole string for comparison.
+  static std::string_view ExtractSuffix(std::string_view domain) {
+    size_t pos = domain.find_last_of('.');
+    return pos != std::string_view::npos ? domain.substr(pos) : domain;
+  }
+
+  std::string_view domain;
+};
+
+namespace internal {
+
+// A custom comparator with strict weak ordering, to allow the lookup of a
+// domain with `DomainSuffixLookup`.
+struct SuffixComparator {
+  using is_transparent = void;
+
+  constexpr bool operator()(const std::string_view& lhs,
+                            const std::string_view& rhs) const {
+    return lhs < rhs;
+  }
+
+  constexpr bool operator()(const std::string_view& lhs,
+                            DomainSuffixLookup rhs) const {
+    return (*this)(lhs, rhs.domain);
+  }
+
+  constexpr bool operator()(DomainSuffixLookup lhs,
+                            const std::string_view& rhs) const {
+    return (*this)(lhs.domain, rhs);
+  }
+};
+
+}  // namespace internal
+
+inline constexpr auto kUnstoppableDomains =
+    base::MakeFixedFlatSet<std::string_view>(
+        {".crypto",     ".x",          ".nft",     ".dao",         ".wallet",
+         ".blockchain", ".bitcoin",    ".zil",     ".altimist",    ".anime",
+         ".klever",     ".manga",      ".polygon", ".unstoppable", ".pudgy",
+         ".tball",      ".stepn",      ".secret",  ".raiin",       ".pog",
+         ".clay",       ".metropolis", ".witg",    ".ubu",         ".kryptic",
+         ".farms",      ".dfz",        ".kresus",  ".binanceus",   ".austin",
+         ".bitget",     ".wrkx"},
+        internal::SuffixComparator());
 
-inline constexpr char kEthDomain[] = ".eth";
-inline constexpr char kDNSForEthDomain[] = ".eth.link";
+inline constexpr std::string_view kEthDomain = ".eth";
+inline constexpr std::string_view kDNSForEthDomain = ".eth.link";
 
-inline constexpr char kSolDomain[] = ".sol";
+inline constexpr std::string_view kSolDomain = ".sol";
```

Comment:
nit: in a global namespace variables usually go before classes/structs.

Reply:
Fixed.